README.md

```diff
@@ -71,6 +80,81 @@ message Transaction {
 }
 ```
 
+### Generating Code
+
+Both `buf` and `protoc` provide valuable utilities for working with protovalidate. While `buf` offers modern features
+and easier integration with various platforms, `protoc` remains a versatile tool for direct code generation.
+Experimenting with both will help you determine which is best suited to your project.
+
+#### `buf`
+
+The `buf` tool facilitates the process of working with protobuf by providing functionality like initialization,
+linting, and code generation. Here are the steps:
+
+1. **Initialize a New Configuration File**: To start a new project, initialize the `buf.yaml` configuration file. This
+   file will define the behavior of the `buf` tool for your project.
+
+   ```shell
+   buf mod init
+   ```
+
+2. **Define the Module Configuration**: The `buf.yaml` file defines a module and is placed at the root of the Protobuf
+   source files. This file includes lint rules, breaking change detection rules, module name, and dependencies. Here's
+   an example of what a typical `buf.yaml` file might look like:
+
+   ```yaml
+   # buf.yaml
+   version: v1
+   # <snip>
+   deps: 
+     - buf.build/bufbuild/protovalidate
+   # <snip>
+   ```
+
+3. **Update Module Dependencies**: You can update a module's dependencies using the `buf.lock` file, which keeps track
+   of the specific versions of dependencies used in the project.
+
+   ```shell
+   buf mod update
+   ```
+
+4. **Define a Local Plugin Template**: The `buf.gen.yaml` file is used to define how code will be generated for specific
+   languages. 
+
+   ```yaml
+   # buf.gen.yaml
+   version: v1
+   # <snip>
+   plugins:
+     - plugin: buf.build/protocolbuffers/python:v23.4
+       out: gen
```

Comment:
doesn't this need an option to generate included protos too?

Reply:
actually it looks like you need https://buf.build/docs/reference/cli/buf/generate/#include-imports

Reply:
looks like you either need to use https://buf.build/docs/reference/cli/buf/generate/#include-imports or manually generate the files using https://github.com/bufbuild/protovalidate-python/blob/main/Makefile#L39